midx.c

```diff
@@ -1199,30 +1219,42 @@ int expire_midx_packs(struct repository *r, const char *object_dir, unsigned fla
 		count[pack_int_id]++;
 	}
 
+	if (flags & MIDX_PROGRESS)
+		progress = start_progress(_("Finding and deleting unreferenced packfiles"),
+				 	 m->num_packs);
 	for (i = 0; i < m->num_packs; i++) {
 		char *pack_name;
 
-		if (count[i])
+		if (count[i]) {
+			display_progress(progress, i + 1);
```

Comment:
Should we just put a `display_progress()` at the beginning of the loop?

Reply:
I think that would definitely be cleaner, I'll go with that approach.  Do you think it makes a difference if I use `i` or `i+1` as the progress there?